### PR TITLE
Adding source code button for documentation.

### PR DESCRIPTION
**Pull request by bruAristimunha**
Adding an option in sphinx to generate page with source code. It solves the issue: #379

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
Thanks 🙏 @bruAristimunha 

The ci failures are I think unrelated. Would you have time to look into it ? 🙏

**Comment by bruAristimunha** [comment]
Yes professor @agramfort, I can look!

**Comment by bruAristimunha** [comment]
Hello @agramfort!

I looked into the code. One error was related to a small difference in the numeric precision. It probably has been caused by the difference between OS or a macro updated in PyTorch or another library. I increased the threshold from 0.1E-3 to 0.5E-2.

The second and third errors involve implementing a filter bank in the module preprocessing.

In the line [469]( https://github.com/braindecode/braindecode/blob/5b4d8daee842c72fbd463b9d4528957fcd32f7be/braindecode/preprocessing/preprocess.py#L469) there is an overwriting of the info values of a mne object. This action tries to preserve the value of the high and low bands. This type of access is not allowed anymore, according to the documentation of the mne. 
image

I recreated all the information from the data object and redefined the value.

**Comment by agramfort** [comment]
@bruAristimunha I let you investigate the CI issues. 🙏

**Comment by bruAristimunha** [comment]
@agramfort, I think everything works.

**Comment by robintibor** [comment]
Thanks for the effort @bruAristimunha , @gemeinl  what do you think about comment above? If we think it's fine we could merge I guess?

**Review by agramfort — APPROVED** [review]
Thx @bruAristimunha 

@robintibor @gemeinl merge if happy

**Comment by bruAristimunha** [comment]
> Thanks for the effort @bruAristimunha , @gemeinl what do you think about comment above? If we think it's fine we could merge I guess?

I thank you for the library. If there's anything I can do to contribute more seriously to the library, I'm super available!

File: braindecode/preprocessing/preprocess.py [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-
 import numpy as np
 import pandas as pd
-from sklearn.utils import deprecated
+from collections.abc import Iterable
+from functools import partial
 from joblib import Parallel, delayed
+from mne import create_info
+from sklearn.utils import deprecated
+from warnings import warn
```
Comment (agramfort) [inline braindecode/preprocessing/preprocess.py:18]: can you just revert the reordering of the imports? we use the convention to import standard lib first, then core packages (numpy, torch) and then local imports

**Comment by agramfort** [comment]
@bruAristimunha what is your status? are you currently a student?

**Comment by bruAristimunha** [comment]
I am a second year PhD student in Computer Science, I would love to collaborate with the library and the projects.

**Comment by agramfort** [comment]
Thx @bruAristimunha 

Your next PRs will be more than welcome too 🙏

**Comment by robintibor** [comment]
Great thanks so much @bruAristimunha :)

**Pull request merged**